<launch>

  <arg name="model" default="quadrotor" />
  <include file="$(find quadrotor_simulator)/launch/rviz.launch"/>
  <include file="$(find korg_nanokontrol)/ground_station.launch"/>

  <arg name="mass" default="0.25"/>
    <include file="$(find mav_manager)/launch/manager.launch"/>

  <group ns="$(arg model)">  
   
   <param name="mass" value="$(arg mass)"/>
   
    <node pkg="quadrotor_simulator"
      type="quadrotor_simulator_so3"
      name="quadrotor_simulator_so3"
      output="screen">
      <param name="rate/odom" value="100.0"/>
      <remap from="~odom" to="odom"/>
      <remap from="~cmd" to="so3_cmd"/>
      <rosparam file="$(find quadrotor_simulator)/config/dynamic_paramsQC2.yaml"/>
    </node>

    <node pkg="nodelet"
      type="nodelet"
      name="standalone_nodelet"
      args="manager"
      output="screen"/>

    <node pkg="nodelet"
      type="nodelet"
      name="trackers_manager"
      args="load trackers_manager/TrackersManager standalone_nodelet"
      output="screen">
      <rosparam file="$(find trackers_manager)/config/trackers.yaml"/>
      <rosparam file="$(find state_control_vehicle)/config/QuadrotorQ2_gains.yaml"/>
      <param name="line_tracker/default_v_des" value="2.0"/>
      <param name="line_tracker/default_a_des" value="1.0"/>
      <remap from="~odom" to="odom"/>
      <remap from="~cmd" to="position_cmd"/>
    </node>

    <node pkg="nodelet"
      type="nodelet"
      args="load so3_control/SO3ControlNodelet standalone_nodelet"
      name="so3_control"
      required="true"
      output="screen">
      <rosparam file="$(find state_control_vehicle)/config/QuadrotorQ2_gains.yaml"/>
      <param name="mass" value="0.25"/>
      <remap from="~odom" to="odom"/>
      <remap from="~position_cmd" to="position_cmd"/>
      <remap from="~so3_cmd" to="so3_cmd"/>
      <remap from="~motors" to="motors"/>
    </node>

    <node pkg="mesh_visualization"
      type="mesh_visualization"
      name="mesh_visualization"
      output="screen">
      <param name="mesh_resource" value="package://mesh_visualization/mesh/hummingbird.mesh"/>
      <param name="color/r" value="0.0"/>
      <param name="color/g" value="0.0"/>
      <param name="color/b" value="1.0"/>
      <param name="color/a" value="0.7"/>
      <remap from="~input" to="odom"/>
    </node>

    <node pkg="state_control_vehicle"
      type="window_visualizer_env"
      name="window_visualizer_env"
      output="screen">
      <rosparam file="$(find state_control_vehicle)/config/env_calib.yaml" />
      </node>
      
    <node pkg="state_control_vehicle"
      type="state_control"
      name="state_control"
      output="screen">
      <param name="traj_filename" type="string" value="/home/giuseppe/Dropbox/papers/2016/icra2017_eagle/trajectory_planning/window_trajectory/traj.txt"/>
      <rosparam file="$(find state_control_vehicle)/config/mav_manager_params.yaml" />
      <remap from="~odom" to="odom"/>
      <remap from="odom" to="odom"/>
      </node>
      
          <!--<include file="$(find mav_manager)/launch/manager.launch"/>-->

  </group>


  </launch>
